app/src/editor/canvas/components/ModuleRenderer/useCanvas.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+
+import { createContext, useContext, type Context, useMemo } from 'react'
+import { RunStates } from '../../state'
+
+type Canvas = {
+    id: ?string,
+    modules: Array<any>,
+    state: $Values<typeof RunStates>,
+}
+
+type CanvasManifest = {
+    canvas: Canvas,
+    isAdjustable: boolean,
+    isEditable: boolean,
+}
+
+type UseCanvasHook = CanvasManifest & {
+    isRunning: boolean,
+}
+
+export const CanvasContext = (createContext({
+    isAdjustable: false,
+    isEditable: false,
+    canvas: {
+        id: null,
+        state: RunStates.Stopped,
+        modules: [],
+    },
+}): Context<CanvasManifest>)
+
+export default (): UseCanvasHook => {
+    const { canvas, isEditable, isAdjustable } = useContext(CanvasContext)
+    const isRunning = canvas.state === RunStates.Running
+
+    return useMemo(() => ({
+        canvas,
+        isAdjustable,
+        isEditable,
+        isRunning,
+    }), [
+        canvas,
+        isAdjustable,
+        isEditable,
+        isRunning,
+    ])
+}
```

Comment:
I like this, though note we already have a `useCanvas` over in `CanvasController`. Maybe these can be unified somehow.

Reply:
Yep, I'll look at the other one and see if I can unify. Fair point.

Reply:
Hm, I predict might have some trouble because of the chain of dependencies required to figure out isAdjustable, isEditable, isRunning. They need stuff from permissions and also some spaghetti state that lives in the RunController.